crates/adapters/src/transport/datagen.rs

```diff
@@ -359,6 +360,14 @@ impl RecordGenerator {
         obj: &mut Value,
     ) -> AnyResult<()> {
         let map = obj.as_object_mut().unwrap();
+        for key in settings.keys() {
+            if !fields.iter().any(|f| f.name.eq(key)) {
```

Comment:
Do you handle duplicate keys as well?

Reply:
No but there isn't much I can do from my side, AFAICT serde will just use the last defined key when deserializing it into a Map